lang/src/main/kotlin/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -443,22 +443,25 @@ fun ExprValue.cast(
             when (targetType) {
                 is BoolType -> when {
                     type.isNumber -> return when {
-                        numberValue().compareTo(0L) == 0 -> valueFactory.newBoolean(false)
-                        else -> valueFactory.newBoolean(true)
+                        numberValue().compareTo(0L) == 0 -> ExprValue.newBoolean(false)
+                        else -> ExprValue.newBoolean(true)
                     }
                     type.isText -> return when (stringValue().toLowerCase()) {
-                        "true" -> valueFactory.newBoolean(true)
-                        "false" -> valueFactory.newBoolean(false)
+                        "true" -> ExprValue.newBoolean(true)
+                        "false" -> ExprValue.newBoolean(false)
                         else -> castFailedErr("can't convert string value to BOOL", internal = false)
                     }
                 }
                 is IntType -> when {
                     type == ExprValueType.BOOL -> return if (booleanValue()) 1L.exprValue(targetType) else 0L.exprValue(targetType)
                     type.isNumber -> return numberValue().exprValue(targetType)
                     type.isText -> {
+                        // Here, we use ion java library to help the transform from string to int
+                        // See if we can get rid of ion here.
```

Comment:
You mean put the code here to a method `parseToLong()`?

Reply:
I can cut a ticket once we merge this PR.

Reply:
More like `parseToLong` now and a ticket for improving it by getting rid of IonValue.  
Isolating the issue into `parseToLong` will save one who picks up the ticket from deciphering the context here.  It will make the ticket more clear and easier to write as well.

Reply:
Do you think a local function works?

Reply:
Something like this?
```
                    type.isText -> {
                        // Here, we use ion java library to help the transform from string to int
                        // TODO: have our own parser implemented and remove dependency on Ion
                        fun parseToLong(s: String): Long {
                            val ion = IonSystemBuilder.standard().build()
                            val value = try {
                                val normalized = stringValue().normalizeForCastToInt()
                                ion.singleValue(normalized) as IonInt
                            } catch (e: Exception) {
                                castFailedErr("can't convert string value to INT", internal = false, cause = e)
                            }
                            return when (value.integerSize) {
                                // Our numbers comparison machinery does not handle big integers yet, fail fast
                                IntegerSize.BIG_INTEGER -> errIntOverflow(8, errorContextFrom(locationMeta))
                                else -> value.longValue()
                            }
                        }

                        return parseToLong(stringValue()).exprValue(targetType)
                    }
```

Reply:
Fine for now, but please refer to the ticket from the in-code comment.

Reply:
https://github.com/partiql/partiql-lang-kotlin/issues/956